src/vdom/builder/after_mount.rs

```diff
@@ -0,0 +1,89 @@
+use crate::{dom_types::View, orders::OrdersContainer, routing::Url};
+
+/// Used for handling initial routing.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum UrlHandling {
+    PassToRoutes,
+    None,
+    // todo: Expand later, as-required
+}
+
+impl Default for UrlHandling {
+    fn default() -> Self {
+        Self::PassToRoutes
+    }
+}
+
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct AfterMount<Mdl> {
+    /// Initial model to be used when the app begins.
+    pub model: Mdl,
+    /// How to handle initial url routing. Defaults to [`UrlHandling::PassToRoutes`] in the
+    /// constructors.
+    pub url_handling: UrlHandling,
+}
+
+impl<Mdl> AfterMount<Mdl> {
+    pub fn new(model: Mdl) -> Self {
+        Self {
+            model,
+            url_handling: UrlHandling::default(),
+        }
+    }
+
+    // TODO: Change to const fn when possible.
+    // TODO: Relevant issue: https://github.com/rust-lang/rust/issues/60964
+    #[allow(clippy::missing_const_for_fn)]
+    pub fn model<NewMdl>(self, model: NewMdl) -> AfterMount<NewMdl> {
+        AfterMount {
+            model,
+            url_handling: self.url_handling,
+        }
+    }
+
+    pub const fn url_handling(mut self, url_handling: UrlHandling) -> Self {
+        self.url_handling = url_handling;
+        self
+    }
+}
+
+pub trait Into<Ms: 'static, Mdl, ElC: View<Ms>, GMs> {
+    fn into_after_mount(
+        self: Box<Self>,
+        init_url: Url,
+        orders: &mut OrdersContainer<Ms, Mdl, ElC, GMs>,
+    ) -> AfterMount<Mdl>;
+}
```

Comment:
The conversion function has a few additional parameters, so I can't directly use `From` or `Into`. I can do something like `Into<(T, Url, &mut OrdersContainer<Ms, Mdl, ElC, GMs>)>`, but then it would be difficult to simply use one of the parameters (`T`) in the types, which is needed elsewhere. I'm not entirely sure how to express `allow this type, T, in the type parameters when used in conjunction with `some_long_list` of other types outside of creating a new trait.

I actually wanted to use `IntoAfterMount`, but clippy complained at me, so I ended up just deleting the trailing words instead of stopping clippy from complaining and re-exported it at a different level (`builder.rs`).

Reply:
> I actually wanted to use IntoAfterMount, but clippy complained at me

We both want to name it `IntoAfterMount` so I suggest to rename it and to allow clippy warning for this trait (and for other similar traits).